client/components/planning-editor-standalone/field-adapters/custom-vocabularies.ts

```diff
@@ -0,0 +1,69 @@
+import {IDropdownConfigVocabulary, IAuthoringFieldV2, ISubject, IVocabularyItem} from 'superdesk-api';
+import {superdeskApi} from '../../../superdeskApi';
+import {IFieldDefinition} from '../profile';
+import {getProfileFieldsConverted} from '../profile-converter';
+
+export const getCustomVocabularyFields = () => {
+    const customVocabularyIds = getProfileFieldsConverted()
+        .filter((x) => x.type === 'custom_vocabulary')
+        .map(({vocabularyId}) => vocabularyId);
+    const result: Array<IFieldDefinition> = [];
+
+    if ((customVocabularyIds?.length ?? 0) > 0) {
+        const allVocabularies = superdeskApi.entities.vocabulary.getAll();
+
+        for (const id of customVocabularyIds) {
+            const vocabulary = allVocabularies.get(id);
+
+            result.push({
+                fieldId: id,
+                getField: ({required, id: _id}) => {
+                    const fieldConfig: IDropdownConfigVocabulary = {
+                        source: 'vocabulary',
+                        vocabularyId: id,
+                        multiple: true,
+                        required: required,
+                    };
+
+                    const field: IAuthoringFieldV2 = {
+                        id: _id,
+                        name: vocabulary.display_name,
+                        fieldType: 'dropdown',
+                        fieldConfig: fieldConfig,
+                    };
+
+                    return field;
+                },
+                storageAdapter: {
+                    storeValue: (item, operationalValue: IVocabularyItem['qcode']) => {
+                        const vocabulary = allVocabularies.get(id);
+                        const vocabItems = vocabulary.items.filter((x) => operationalValue?.includes(x.qcode)) ?? [];
```

Comment:
only return a boolean from a `.filter` callback. Nullish coalescing would be nicer here than optional chaining here.